rero_ils/modules/patron_transactions/api.py

```diff
@@ -106,6 +106,24 @@ def get_transactions_pids_for_patron(cls, patron_pid, status=None):
         for result in results:
             yield result.pid
 
+    @classmethod
+    def get_transactions_total_amount_for_patron(cls, patron_pid, status=None,
+                                                 with_subscription=True):
+        """Get total amount transactions linked to a patron.
+
+        :param patron_pid: the patron pid being searched
+        :param status: (optional) transaction status filter,
+        :param with_subscription: (optional) include or exclude subscription
+        type filter,
+        """
+        query = cls._build_transaction_query(patron_pid, status)
+        if not with_subscription:
+            query = query.exclude('terms', type=['subscription'])
+        query.aggs.metric('transactions_total_amount',
+                          'sum', field='total_amount')
+        results = query.execute()
```

Comment:
I'm not sure that you need to execute you can use bucket I think: https://elasticsearch-dsl.readthedocs.io/en/latest/search_dsl.html#aggregations or at least put the size to 0.

Reply:
ok